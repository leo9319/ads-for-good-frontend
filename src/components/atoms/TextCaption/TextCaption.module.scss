@import '../../../styles/colours';

.textCaption {
  margin: 0;

  &.error {
    color: $color-red-600;
  }

  &.success {
    color: $color-green-800;
  }
}
