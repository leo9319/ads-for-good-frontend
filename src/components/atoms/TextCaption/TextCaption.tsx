import React, { ReactElement } from 'react';

import { classNames, getModuleClasses } from '@utils/common/classNames';
import Text from '@radix-styles/atoms/Text';

import styles from './TextCaption.module.scss';
import { TextSize, TextWeight } from '@internal/types/common/style';

/**
 * @interface
 * The TextCaptionProps type of {@link TextCaption}.
 */
export interface TextCaptionProps {
  /**
   * The messages to be shown, it can be a string or an array of strings
   * @property
   */
  messages?: string | string[] | null;
  /**
   * The mode of the text caption to show the success or error messages
   * @property
   */
  mode?: 'success' | 'error';
  /**
   * The size of the text caption
   * @property
   */
  size?: TextSize;
  /**
   * The weight of the text caption
   * @property
   */
  weight?: TextWeight;
}

/**
 * TextCaption component to show the success or error messages
 *
 * @group Atoms
 * @category Component
 *
 * @param {TextCaptionProps} props - The TextCaption props
 * @returns {ReactElement} - TextCaption Component
 */
export const TextCaption = ({
  messages,
  mode = 'error',
  size = '2',
  weight = 'regular',
}: TextCaptionProps): ReactElement | null => {
  const newClasses = getModuleClasses(mode, styles);
  if (!messages) {
    return null;
  }
  const captions = typeof messages == 'string' ? [messages] : messages;
  return (
    <div className={classNames(styles.textCaption, newClasses)}>
      {captions.map((caption: string, index: number) => (
        <Text size={size} weight={weight} key={caption + '-' + index}>
          {caption}
          <br />
        </Text>
      ))}
    </div>
  );
};

export default TextCaption;
