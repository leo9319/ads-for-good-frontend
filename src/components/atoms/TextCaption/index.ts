export type { TextCaptionProps } from './TextCaption';
export { default } from './TextCaption';
export { TextCaption } from './TextCaption';
