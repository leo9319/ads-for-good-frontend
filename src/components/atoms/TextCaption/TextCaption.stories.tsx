import type { Meta, StoryObj } from '@storybook/react';
import { TextCaption } from './TextCaption';

const meta = {
  title: 'Library/Atoms/TextCaption',
  component: TextCaption,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    messages: {
      control: 'object',
    },
    mode: {
      options: ['success', 'error'],
      control: {
        type: 'select',
      },
    },
    size: {
      control: {
        type: 'select',
      },
      options: [
        '1',
        '2',
        '3',
        '4',
        '5',
        '6',
        '7',
        '8',
        '9',
        '10',
        '11',
        '12',
        '13',
      ],
    },
    weight: {
      control: {
        type: 'select',
      },
      options: ['regular', 'semibold', 'bold'],
    },
  },
  args: {
    mode: 'error',
    size: '2',
    weight: 'regular',
  },
} satisfies Meta<typeof TextCaption>;

export default meta;
type Story = StoryObj<typeof meta>;

export const SingleMessage: Story = {
  args: {
    messages: 'This field is required.',
  },
};

export const MultipleMessages: Story = {
  args: {
    messages: [
      'Invalid email address.',
      'Password must be at least 8 characters.',
    ],
  },
};

export const NoMessage: Story = {
  args: {
    messages: null,
  },
};
