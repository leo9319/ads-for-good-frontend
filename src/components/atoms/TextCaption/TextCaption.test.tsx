import React from 'react';
import TextCaption from './TextCaption';
import { render } from '@testing-library/react';

describe('TextCaption', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<TextCaption messages="Has Error" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should support multiple errors', () => {
    const { baseElement } = render(<TextCaption messages={['Has Error']} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should return empty container when passing the empty data', () => {
    const { baseElement } = render(<TextCaption messages={null} />);
    expect(baseElement).toMatchSnapshot();
  });
});
