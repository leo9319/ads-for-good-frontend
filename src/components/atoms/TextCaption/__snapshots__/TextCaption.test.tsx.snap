// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCaption should render successfully 1`] = `
<body>
  <div>
    <div
      class="textCaption error"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
      >
        Has Error
        <br />
      </span>
    </div>
  </div>
</body>
`;

exports[`TextCaption should return empty container when passing the empty data 1`] = `
<body>
  <div />
</body>
`;

exports[`TextCaption should support multiple errors 1`] = `
<body>
  <div>
    <div
      class="textCaption error"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
      >
        Has Error
        <br />
      </span>
    </div>
  </div>
</body>
`;
